sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/StreamingCheckpointSuite.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.streaming
+
+import java.net.URI
+
+import org.apache.hadoop.fs.{LocalFileSystem, Path, RawLocalFileSystem}
+import org.apache.hadoop.fs.permission.FsPermission
+
+import org.apache.spark.{SparkConf, SparkFunSuite}
+import org.apache.spark.sql.{Encoder, LocalSparkSession, SparkSession, SQLContext}
+
+class StreamingCheckpointSuite extends SparkFunSuite with LocalSparkSession {
+
+  test("temp checkpoint dir should stay local even if default filesystem is not local") {
```

Comment:
I guess having SPARK-26825 as prefix helps to the future readers why the change was necessary. Please ignore of test name is too long to add it.